src/main/java/net/logstash/logback/appender/AbstractLogstashTcpSocketAppender.java

```diff
@@ -375,6 +421,18 @@ private synchronized void openSocket() {
                     
                     this.socket = tempSocket;
                     this.outputStream = tempOutputStream;
+                    
+                    /*
+                     * If connected to a secondary, remember when the connection should be closed to
+                     * force attempt to reconnect to primary
+                     */
+                    if( hostIndex != 0 ) {
```

Comment:
_nitpick_: This would read a little better (and be more self-documenting) if you used a constant, or created a wrapper method.

`if (hostIndex != PRIMARY_HOST_INDEX) {`

or 

`if (isPrimaryHost(hostIndex)) {`